DepositRegistrar.sol

```diff
@@ -0,0 +1,306 @@
+            name.deed = sealedBids[seal];
+        
+        } else if (bid.value() > name.value) {
+            // not winner, but affects second place
+            name.value = bid.value();
+            bid.closeDeed(1);
+            
+        } else {
+            // bid doesn't affect auction
+            bid.closeDeed(1);
+        }
+    }
+    
+    function finalizeAuction(bytes32 nameHash) noEther {
+        entry name = hashedNames[nameHash];
+        if (now < name.auctionDate || name.status != Mode.Auction) throw;
+        
+        // set the name
+        name.status = Mode.Owned;
+        name.firstRegistered = now;
+        name.lastRenewed = now;
+
+        //set the averages
+        uint period = (now - lastSinceNewName) * M;
+        averagePeriod = (9999 * averagePeriod + period) / 10000;
+        uint n = averagePeriod < (60 seconds * M) ? 7 days * M / averagePeriod : 10000;
+        averagePrice = ((n-1)*averagePrice + name.value ) / n;
+        name.averagePrice = averagePrice;
+        lastSinceNewName = now;
+        
+        Deed deedContract = Deed(name.deed);
+        
+        deedContract.setBalance(name.value > minBid ? name.value : minBid);
+    }
+    
+    
+    /*
+    ## Renewals can be done at any moment by renewing the deposit
+
+    In order to renew a name you need to calculate how much have average prices of 
+    names changed since you last bought it. If the average price names have increased, 
+    then you need to increase your deposit. If median prices have fallen down, then 
+    you have the right to withdraw part of the difference (1/2 after a year, 3/4 
+    after 2 years, 7/8 after 3 years etc). 
+    */ 
+    function updatedValue(bytes32 nameHash) constant returns (uint updatedPrice) {
+        entry name = hashedNames[nameHash];
+        return name.value * averagePrice / name.averagePrice;
+    }
+    
+    function renewDeed(bytes32 nameHash) {
+        entry name = hashedNames[nameHash];
+        if (now < name.firstRegistered + renewalPeriod/2 ) throw;
+        Deed deedContract = Deed(name.deed);
+        uint difference = 0;
+        if (msg.sender != deedContract.owner() && name.status != Mode.Owned) throw;
+
+        uint updatedPrice = updatedValue(nameHash);
+        
+        if (updatedPrice > name.value) {
+            difference = updatedPrice - name.value; 
+            if (msg.value < difference) throw;
+            deedContract.setBalance.value(msg.value)(updatedPrice);
+        } else {
+            difference =  name.value - updatedPrice; 
+            uint ratioOfRecovery = 2**((now - name.lastRenewed)/ 1 years);
+            // the more time, the more you recover more from your initial investment
+            deedContract.setBalance(name.value + difference - difference/ratioOfRecovery);
+        }
+        
+        name.value = updatedPrice;
+        name.lastRenewed = now;
+        // next renewal data is twice the current age
```

Comment:
one year goes rather quickly. I hate to renew domains every year, specially if it's a hassle and you own many. This way you have to renew after 1 year, then after 2 years, 4 etc. Maybe we can add a upper limit, someone only needing to renew it after 8 or 16 years might be a stretch, but hey, maybe when ethereum has existed as long as the web, it might make sense.